src/PopChainManager/index.ts

```diff
@@ -11,27 +11,37 @@ export class PopChainManager {
   public getFullPopChain(pop: Pop, hideFullChain: boolean): Pop[] {
     let popChain: Pop[] = [];
 
-    if (!!pop.childPops.length) {
-      pop.childPops.forEach(function(child: Pop): void {
-        let idx = child.parentPop.childPops.indexOf(child);
-        if (idx !== -1) {
-          child.parentPop.childPops.splice(idx, 1);
-        }
-        child.parentPop = null;
-        popChain = popChain.concat(this.getFullPopChain(child, hideFullChain));
-      }, this);
+    if (hideFullChain) {
+      pop = this._getRootPop(pop);
+    } else {
+      pop = this._getFirstUnpinnedParentPop(pop);
     }
 
-    popChain.push(pop);
+    let stack: Pop[] = [];
+    stack.push(pop);
 
-    if (!!pop.parentPop) {
-      let idx = pop.parentPop.childPops.indexOf(pop);
-      if (idx !== -1) {
-        pop.parentPop.childPops.splice(idx, 1);
+    while (!!stack.length) {
+      pop = stack.pop();
+
+      if (!!pop.parentPop) {
+        let idx = pop.parentPop.childPops.indexOf(pop);
+        if (idx !== -1) {
```

Comment:
I am a little skeptical about if this if-condition could be hit, but I guess its okay as a precautionary measure. Might be worth throwing a warning if somehow `idx === -1`